### PR TITLE
Knowledge lookup prompts:

### PR DESCRIPTION
**Pull request by olehborysevych**
Knowledge lookup golden prompts for Duet AI
1.  Kafka IO
2. Bigquery IO
3. Bigtable IO
4. Spanner IO
5. Json 
6. CSV
7. Avro
8. Parquet
9. Jdbc

Code Explanation prompts
1.  Kafka IO
2. Bigquery IO
3. Bigtable IO
4. Spanner IO
5. Json 
6. CSV
7. Avro
8. Parquet
9. Jdbc
10. 
Code Generation prompts
1.  Kafka IO
2. Bigquery IO
3. Bigtable IO
4. Spanner IO
5. Json 
6. CSV
7. Avro
8. Parquet
9. Jdbc

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: learning/prompts/documentation-lookup/18_io_kafka.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Prompt:
+Is Apache Kafka supported in Apache Beam?
+Response:
+[Apache Kafka](https://kafka.apache.org/) is a distributed streaming platform that is used for building real-time data pipelines and streaming applications. It is horizontally scalable, fault-tolerant, and fast.
+Apache Beam has a `KafkaIO` built-in connector for Apache Kafka that allows you to read and write data from and to Kafka. Apache kafka is supported both in batch and streaming pipelines and as a source and a sink.
```
Comment (alxp1982) [inline learning/prompts/documentation-lookup/18_io_kafka.md:5]: Apache Beam has a `KafkaIO` built-in connector for Apache Kafka that allows you to read and write data from and to Kafka. Apache Kafka is supported both in batch and streaming pipelines and as a source and a sink.

File: learning/prompts/documentation-lookup/21_io_bigtable.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+Prompt:
+Is Cloud BigTable supported in Apache Beam?
+Response:
+[Cloud Bigtable](https://cloud.google.com/bigtable) is a [Google Cloud](https://cloud.google.com/) service that provides  enterprise-grade NoSQL database service with single-digit millisecond latency, limitless scale, and 99.999% availability for large analytical and operational workloads.
```
Comment (alxp1982) [inline learning/prompts/documentation-lookup/21_io_bigtable.md:4]: [Cloud Bigtable](https://cloud.google.com/bigtable) is a [Google Cloud](https://cloud.google.com/) service that provides enterprise-grade NoSQL database service with single-digit millisecond latency, limitless scale, and 99.999% availability for large analytical and operational workloads.

File: learning/prompts/documentation-lookup/23_io_gcs.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Prompt:
+Is GCS file system supported in Apache Beam?
+Response:
+[Google Cloud Storage](https://cloud.google.com/storage) is a popular object storage service for storing and accessing data on Google Cloud. Apache Beam supports both working with GCS file system and reading and writing data from and to GCS buckets.
```
Comment (alxp1982) [inline learning/prompts/documentation-lookup/23_io_gcs.md:4]: [Google Cloud Storage](https://cloud.google.com/storage) is a popular object storage service for storing and accessing data on Google Cloud. Apache Beam supports both working with the GCS file system and reading and writing data from and to GCS buckets.

File: learning/prompts/documentation-lookup/23_io_gcs.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+* [Python](https://beam.apache.org/releases/pydoc/current/apache_beam.io.textio.html)
+* [Go](https://pkg.go.dev/github.com/apache/beam/sdks/v2/go/pkg/beam/io/textio)
+* [Typescript via X-language](https://beam.apache.org/documentation/sdks/typescript/io/textio/)
+
+Reading and writing data in different file formats from and to GCS buckets is also supported in the AvroIO, XMLIO, TFRecordIO, and ParquetIO connectors.
+
+GCS path is specified using the following format: `gs://<bucket>/<path>`. For example, `gs://my-bucket/my-file.txt`.
+Read and write transforms support reading and writing data from and to multiple files using wildcards. For example, `gs://my-bucket/my-files-*.txt`.
+
+Apache Beam also has a support of GCS file system that allows you work with GCS file system directly. For example, you can use GCS file system to check if a file exists, list files in a directory, or delete a file. GCS file system is supported in the following languages:
```
Comment (alxp1982) [inline learning/prompts/documentation-lookup/23_io_gcs.md:17]: Apache Beam also has support for the GCS file system that allows you to work with the GCS file system directly. For example, you can use GCS file system to check if a file exists, list files in a directory, or delete a file. GCS file system is supported in the following languages:

**Review by alxp1982 — CHANGES_REQUESTED** [review]
Please address the feedback

**Comment by olehborysevych** [comment]
@alxp1982 your comments are resolved

**Comment by johnjcasey** [comment]
Why are these prompts getting merged back into Beam?

**Review by alxp1982 — APPROVED** [review]

**Comment by liferoad** [comment]
R: @rszper

**Comment by rszper** [comment]
Adding Sereana from the TW team as a reviewer, because she's helping with these this quarter.

R: @sereanaseim

**Review by sereanaseim — COMMENTED** [review]
There's a typo in the code explanation folder name - it should be `code-explanation`, without the extra i.

Sorry for the volume of suggested edits. GitHub isn't great for proofreading documentation. The edits should be straightforward, though. Let me know if you have any questions.

**Comment by olehborysevych** [comment]
Thank you, @sereanaseim I applied your changes. Could you please take a look?

File: learning/prompts/documentation-lookup/18_io_kafka.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+Prompt:
+Is Apache Kafka supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/18_io_kafka.md:3]: Added a paragraph break
```suggestion
Is Apache Kafka supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/21_io_bigtable.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+Prompt:
+Is Cloud BigTable supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/21_io_bigtable.md:3]: Added a paragraph break
```suggestion
Is Cloud BigTable supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/22_io_spanner.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+Prompt:
+Is Cloud Spanner supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/22_io_spanner.md:3]: Added a paragraph break
```suggestion
Is Cloud Spanner supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/22_io_spanner.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+[Cloud Spanner](https://cloud.google.com/spanner) is a [Google Cloud](https://cloud.google.com/) service that provides a fully managed, mission-critical, relational database service that offers transactional consistency at global scale, schemas, SQL (ANSI 2011 with extensions), and automatic, synchronous replication for high availability.
+
+Apache Beam has a built-in SpannerIO connector for Cloud Spanner that lets you read data from and write data to Cloud Spanner. Cloud Spanner is supported in both batch pipelines and streaming pipelines. and as a source and a sink. For more information, see the SpannerIO connector documentation:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/22_io_spanner.md:6]:
```suggestion
Apache Beam has a built-in SpannerIO connector for Cloud Spanner that lets you read data from and write data to Cloud Spanner. Cloud Spanner is supported in both batch pipelines and streaming pipelines, and as a source and a sink. For more information, see the SpannerIO connector documentation:
```

File: learning/prompts/documentation-lookup/23_io_gcs.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
+Prompt:
+Is Google Cloud Storage file system supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/23_io_gcs.md:3]: Added a paragraph break
```suggestion
Is Google Cloud Storage file system supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/24_io_json.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
+Prompt:
+Is JSON supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/24_io_json.md:3]: Added a paragraph break
```suggestion
Is JSON supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/25_io_csv.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
+Prompt:
+Is CSV format supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/25_io_csv.md:3]: Added a paragraph break
```suggestion
Is CSV format supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/26_io_avro.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
+Prompt:
+Is Apache Avro format supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/26_io_avro.md:3]: Added a paragraph break
```suggestion
Is Apache Avro format supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/27_io_parquet.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
+Prompt:
+Is Apache Parquet format supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/27_io_parquet.md:3]: Added a paragraph break
```suggestion
Is Apache Parquet format supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/28_io_jdbc.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
+Prompt:
+Is Oracle Java Database Connectivity (JDBC) supported in Apache Beam?
+Response:
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/28_io_jdbc.md:3]: Added a paragraph break
```suggestion
Is Oracle Java Database Connectivity (JDBC) supported in Apache Beam?

Response:
```

File: learning/prompts/documentation-lookup/28_io_jdbc.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+The [Oracle Java Database Connectivity (JDBC) API](https://docs.oracle.com/javase/8/docs/technotes/guides/jdbc/) is a Java API for connecting to relational databases. Apache Beam supports reading and writing data with JDBC using the `ReadFromJdbc` and `WriteToJdbc` transforms in the `JdbcIO` module. For more information, see the JdbcIO connector documentation:
+* [Java](https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/io/jdbc/JdbcIO.html),
+* [Python (using cross-language transforms)](https://beam.apache.org/releases/pydoc/current/apache_beam.io.jdbc.html).
```
Comment (sereanaseim) [inline learning/prompts/documentation-lookup/28_io_jdbc.md:6]:
```suggestion
* [Java](https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/io/jdbc/JdbcIO.html)
* [Python (using cross-language transforms)](https://beam.apache.org/releases/pydoc/current/apache_beam.io.jdbc.html)
```

**Review by sereanaseim — COMMENTED** [review]
Just a few more minor changes for grammar and formatting. Thanks!

**Comment by olehborysevych** [comment]
HI @sereanaseim  Thank you for your attention and guidance. Your comments are addressed.

**Review by damccorm — APPROVED** [review]
LGTM, thanks

**Pull request merged**